### ISSUE TITLE
Overview of unique plugins instead of available (duplicate) plugins

### ISSUE THREAD

**Issue by Zawadidone** [fox-it/dissect.target#822]
I would like to have an option that only shows unique plugins, e.g. `target-query --list-unique`

* That will not show namespace plugins like `browser.*`, `webserver.*`, etc.
* That will not show alias plugins like `bashhistory`.
* That will show unique plugins names for every plugin, e.g. `os.unix.bsd.citrix.history.commandhistory` vs `os.unix.history.commandhistory`.

The problems I have with the current overview when executing `target-query -l` are explained below.

## Namespace plugins

```
[...]
browser:
  brave:
    brave.cookies - Return browser cookie records for Brave. (output: records)
    brave.downloads - Return browser download records for Brave. (output: records)
    brave.extensions - Return browser extension records for Brave. (output: records)
    brave.history - Return browser history records for Brave. (output: records)
    brave.passwords - Return browser password records for Brave. (output: records)
    get_all_records - Return the records of all exported methods. (output: records)
  browser:
    browser.cookies - Return cookies for: firefox,chromium,edge,brave,chrome (output: records)
    browser.downloads - Return downloads for: firefox,chromium,edge,brave,iexplore,chrome (output: records)
    browser.extensions - Return extensions for: firefox,chromium,edge,brave,chrome (output: records)
    browser.history - Return history for: firefox,chromium,edge,brave,iexplore,chrome (output: records)
    browser.passwords - Return passwords for: firefox,chromium,edge,brave,chrome (output: records)
    get_all_records - Return the records of all exported methods. (output: records)
[...]
```

Is it intended that plugins can be executed in different ways? Based on the available plugins that are shown, e.g. `brave.cookies`:

* `-f brave.cookies`
* `-f brave.get_all_records` or `-f get_all_records`, both functions are shown in the overview but do not work
* `-f browser`
* `-f browser.cookies`

## Plugins that inherit a namespace plugin with the same name

```
[...]
os:
  unix:
    bsd:
      citrix:
        history:
          bashhistory - Return shell history for all users. (output: records)
          commandhistory - Return shell history for all users. (output: records)
          get_all_records - Return the records of all exported methods. (output: records)
    history:
      bashhistory - Return shell history for all users. (output: records)
      commandhistory - Return shell history for all users. (output: records)
[...]
```

Is it intended that the plugin `os.unix.bsd.citrix.history.commandhistory` has the same name as `os.unix.history.commandhistory`?

## References
* https://github.com/fox-it/dissect.target/pull/194
* https://github.com/fox-it/dissect.target/pull/329

**Comment by Schamper** [comment]
Without going into detail for everything yet (I'm on mobile) I'd just like to mention that `get_all_records` is a bug that's addressed with #763. But yes, `brave.cookies`, `browser` and `browser.cookies` actually do three different things. The first only executes `brave.cookies`, the second executes _all_ browser plugins for _all_ browsers, the third executes the cookies plugin for _all_ browsers.

The latter is also intentional. You should view it as just `commandhistory` with a more specific implementation available for Citrix.

The "tree" is not leading in any way for plugin names, it's just an arbitrary folder structure that we try to put some logic into. The actual function name is leading. So both `commandhistory` are actually the same thing, one just has a more OS specific variant of it that's only available for Citrix targets.

Perhaps the current "tree" form is not the most ideal way to list plugins. At the time it was just chosen as it provided a little context, but with the ever growing list of plugins it's just becoming confusing. We've been wanting plugin categories/labeling since pre-1.0, which would perhaps allow a more friendly and easier to understand function listing, separated into _what_ a plugin provides, not by _where_ a plugin lives in the repository.

(That's actually a lot more detail than I intended to go into 😅)

**Comment by Zawadidone** [comment]
Thanks for the explanation!
